packages/flutter_tools/test/general.shard/build_system/targets/assets_test.dart

```diff
@@ -638,3 +623,33 @@ flutter:
     expect(logger.errorText, isEmpty);
   });
 }
+
+/// Write a `.dart_tool/package_config.json` file at [directory].
+///
+/// It will contain a package entry for [mainLibName] with `rootUri` at
+/// [directory].
+///
+/// [otherLibs] maps other package names to their `rootUri`.
+void writePackageConfigFile(
```

Comment:
Should this be moved into test/src and used through test instead of hard-coding a small variant of this across other tests?